bundles/ranvier-combat/behaviors/item/decay.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+module.exports = srcPath => {
+  const ItemType = require(srcPath + 'ItemType');
+  const Logger = require(srcPath + 'Logger');
+
+  let decayStarted,
+      decayEnd = Infinity;
+
+  return {
+    listeners: {
+      updateTick: state => function (config) {
+        let { duration = 60 } = config;
+        duration = duration * 1000;
+        const now = Date.now();
+
+        if (decayStarted) {
+          if (decayEnd < now) {
+            destroyItem(state, this);
+          } else {
+            checkForHalfRotted(this, duration, now);
+          }
+        } else {
+          decayStarted = true;
+          decayEnd = Date.now() + duration;
+        }
+      }
+    }
+  };
+
+  function destroyItem(state, rottedItem) {
+    Logger.verbose(`${rottedItem.id} has decayed.`);
+    state.ItemManager.remove(rottedItem);
```

Comment:
Not until later on, but there's an issue with this due to the behavior noted below, where the item now has a reference to the inventory it is a part of.